compiler/rustc_middle/src/ty/query/plumbing.rs

```diff
@@ -163,7 +166,11 @@ impl<'tcx> TyCtxt<'tcx> {
             }
         });
 
-        eprintln!("end of query stack");
+        if num_frames != None {
+            eprintln!("we're just showing a limited slice of the query stack");
+        } else {
+            eprintln!("end of query stack");
+        }
```

Comment:
I think what bjorn3 is suggesting is

```suggestion
        if num_frames == None || num_frames >= Some(i) {
            eprintln!("end of query stack");
        } else {
            eprintln!("we're just showing a limited slice of the query stack");
        }
```

but you'll also have to pull out `let mut i = 0;` so that we have access to `i` here.

This will make sure `end of query stack` is printed if there are less than `num_frames` frames to print.